File: netlogo-gui/src/main/prim/_returnreport.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.I18N;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.EngineException;

public final strictfp class _returnreport extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax();
  }

  @Override
  public String displayName() {
    return "END"; // for use in error messages
  }

  @Override
  public void perform(Context context) throws LogoException {
    perform_1(context);
  }

  public void perform_1(Context context) throws LogoException {
    throw new EngineException(context, this,
            I18N.errorsJ().get("org.nlogo.prim._returnreport.reportNotCalledInReportProcedure"));
  }
}

File: netlogo-headless/src/main/prim/_returnreport.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.core.I18N
import org.nlogo.nvm.{ Command, Context}
import org.nlogo.nvm.RuntimePrimitiveException

class _returnreport extends Command {
  override def displayName = "END"  // for use in error messages
  override def perform(context: Context): Unit = {
    perform_1(context)
  }
  def perform_1(context: Context): Unit = {
    throw new RuntimePrimitiveException(
      context, this,
      I18N.errors.get(
        "org.nlogo.prim._returnreport.reportNotCalledInReportProcedure"))
  }
}
